src/posts/entities/post.repository.spec.ts

```diff
@@ -0,0 +1,163 @@
+import { PostRepository } from './post.repository';
```

Comment:
I didn't fully get what is happening in this file/tests (too late and I am too exhausted), however, I improved the mocks and the date usage and kept the rest untouched. maybe we can discuss it later